### PR TITLE
cross-platform userdata

### PR DESCRIPTION
**Pull request by bng0y**
- Allow the region to be passed by command line `--region` [OSD-9344](https://issues.redhat.com/browse/OSD-9344)
- Uses [cloud-config](https://cloudinit.readthedocs.io/en/latest/topics/examples.html) syntax for userdata script
- Validating reachability to region specific resource. e.g ec.<region>.amazonaws.com [OSD-9222](https://issues.redhat.com/browse/OSD-9222)

### CODE REVIEW

**Comment by tiwillia** [comment]
One comment, the rest of this LGTM

File: cmd/egress/cmd.go [hunk 1]
```diff
@@ -15,11 +15,23 @@ var defaultTags = map[string]string{
 	"osd-network-verifier": "owned",
 }
 
+func getDefaultRegion() string {
```
[the remark this answers is not in the document]
Comment (bng0y) [reply cmd/egress/cmd.go:18]: interestingly, it shows up in the `./osd-network-verifier --help`
```
--region string               Region to validate (default "us-east-2")
```
or 
`default "AWS_DEFAULT_REGION"` when it is set

**Comment by fahlmant** [comment]
@bng0y Could you please add a description?

File: cmd/egress/cmd.go (hunk 1 above)
[the remark this answers is not in the document]
Comment (bng0y) [reply cmd/egress/cmd.go:20]: I returned the `logger.Warn(ctx, "Using region: %s", region)` and updated the `--help` with more explanation

File: config/userdata.yaml [hunk 2]
```diff
@@ -0,0 +1,13 @@
+#cloud-config
+repo_update: true
+
+packages:
+  - docker 
+
+runcmd:
+  - [ sh, -c, echo "USERDATA BEGIN"]
+  - sudo service docker start
```
Comment (tiwillia) [inline config/userdata.yaml:9]: Just wanted to note that this removes `yum update` and `yum install docker`.

Comment (bng0y) [reply config/userdata.yaml:9]: the `packages: docker` covers the latter.

Comment (bng0y) [reply config/userdata.yaml:9]: we could consider also adding `package_update: true`

Comment (tiwillia) [reply config/userdata.yaml:10]: James Harrington confirmed in slack that we did not need to worry about doing `yum update`, so fine to leave `package_update: true` out fro now I think.

**Comment by tiwillia** [comment]
LGTM. Will conflict with https://github.com/openshift/osd-network-verifier/pull/31 , lets merge this one first imo.

**Review by fahlmant — APPROVED** [review]
/approve

**Pull request merged**